terragrunt/aws/api_gateway/api_gateway.tf

```diff
@@ -0,0 +1,34 @@
+resource "aws_apigatewayv2_api" "cra_upd_apigw_endpoint" {
```

Comment:
Why do you need an API gateway here? The ECS should be enough but maybe I am missing something. Also, is there an architectural diagram by any chance so that we can visualize this better? Thanks!

Reply:
API gateway integrates directly with Cloudfront, which simplifies the integration of Cloudfront with the backend, and allows Cloudfront to act essentially like a reverse proxy. Which, with API gateway, can route the backend requests like:

(WAF) -> Cloudfront -> API Gateway -> (VPC link) -> ECS

With only Cloudfront accessible from the internet, and only through the WAF, which only allows connections from allowed IPs. Which also allows us to avoid having to configure additional ingress rules for the VPC and resources, among other things. API Gateway also gives us an easy pathway for adding Lambda endpoints or integrating separate services, if ever we wanted to do that.

And yes, I'll email you the architecture diagram we have.

Reply:
Thanks for the architecture diagram Marc! I do agree that an API gateway in the future will allow an easy pathway for adding Lambda endpoints or integrating separate services, but it is not needed now and you can always add it in the future. 

Instead of API gateway, wouldn't a load balancer serve you better? I don't think I saw this in the architectural diagram but I think it might better fit your needs and is a standard archtecture model.